libs/api-implementation/orchestration/src/lib/orchestration.service.ts

```diff
@@ -1,108 +1,138 @@
+    this.runningMachines = [];
+    this.machines = {};
+  }
 
-    async createMachine(name: string) {
-        try {
-            const [vm, operation] = await this.createVM(name);
-            this.runningMachines.push(vm);
-            this.numOfTotalInstances++;
-            console.log("Created the following machine: \n" + operation.metadata)
-        } catch (error) {
-            throw (error);
-        }
+  async createMachine(name: string) {
+    try {
+      const [vm, operation] = await this.createVM(name);
+      this.runningMachines.push(vm);
+      this.numOfTotalInstances++;
+      console.log('Created the following machine: \n' + operation.metadata);
+    } catch (error) {
+      throw error;
     }
+  }
+
+  public async createVM(name: string) {
+    let vm, operation;
+    const config = await this.getConfigFile();
+    [vm, operation] = await this.zone.createVM(name, config);
+    this.machines[name] = new VM(name, Status.Running, vm);
+    return [this.machines[name], operation];
+  }
+
+  private async getConfigFile() {
     
-    public async createVM(name: string) {
-        let vm, operation;
-        [vm, operation] = await this.zone.createVM(name, { os: 'ubuntu' });
-        this.machines[name] = new VM(name, Status.Running, vm);
-        return [this.machines[name], operation]    
-    }
+    const script = await doAsync(fs).readFile(
+      path.resolve(__dirname, `../../../script.sh`)
+    );
 
-    async reset(name: string) {
-        try {
-            const response = await this.machines[name].getInstance().reset();
-            console.log(response);
-        } catch (error) {
-            throw (error);
-        }
-    }
+    console.log(script.toString());
 
-    async start(name: string) {
-        try {
-            const response = await this.machines[name].getInstance().start();
-            this.runningMachines.push(this.machines[name]);
-            console.log(response);
-        } catch (error) {
-            throw (error);
-        }
-    }
-    async stop(name: string) {
-        try {
-            const response = await this.machines[name].getInstance().stop();
-            const index = this.runningMachines.indexOf(this.machines[name].getInstance());
-            if (index > -1) {
-                this.runningMachines.splice(index, 1);
-            }
-            console.log(response);
-        } catch (error) {
-            throw (error);
-        }
-    }
-    async deleteMachine(name: string) {
-        try {
-            if (this.machines[name].getStatus() == Status.Running) {
-                const index = this.runningMachines.indexOf(this.machines[name].getInstance());
-                this.runningMachines.splice(index, 1);
-            }
-            this.numOfTotalInstances--;
-            const response = await this.machines[name].getInstance().delete();
-            console.log(response);
-        } catch (error) {
-            throw (error);
-        }
+    return {
+      os: 'ubuntu',
```

Comment:
Do you think it would be better to not hardcode this and add to some constants file in case we change to a different Linux system in the future. Just a recommendation, you can take your call.

Reply:
Yeap, we can do that.